### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                        connection.prepareStatement(
+                             insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            insertAllDrivers(car.getId(), car.getDrivers());
```
Comment (Max-Pochepets) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: when you create a new car, it does not have drivers, drivers are added with a specific method from `CarServise`

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                drivers.add(parseDriverFromResultSet(resultSet));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all drivers", e);
```
Comment (Max-Pochepets) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: I guess we should make exception messages more informative
```suggestion
            throw new DataProcessingException("Can't get all drivers by car id " + carId, e);
```

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                preparedStatement.setLong((i * 2) + 1, carId);
+                preparedStatement.setLong((i * 2) + 2, driver.getId());
+            }
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers " + drivers, e);
+        }
+    }
+
+    private void deleteAllDriversExceptList(Long carId, List<Driver> exceptions) {
```
Comment (Max-Pochepets) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: what does this method do, I don't quite get it?

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+    }
+
+    private void updateDriversByCar(Car car) {
+        insertAllDrivers(car.getId(), car.getDrivers());
+        deleteAllDriversExceptList(car.getId(), car.getDrivers());
+    }
+
+    private void insertAllDrivers(Long carId, List<Driver> drivers) {
```
Comment (Max-Pochepets) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: Im not sure you need to complicate it this much, you don't need to delete or insert ALL drivers, you can do it with a simple query in respective methods, such as `DELETE FROM cars_drivers WHERE car_id = ?`

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                        connection.prepareStatement(
+                                selectQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete driver by id " + id, e);
+        }
+    }
+
+    private Driver parseFromResultSet(ResultSet resultSet) throws SQLException {
```
Comment (Max-Pochepets) [inline src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java:105]: you can make this method `package-private static` and use this one in all your daos to not duplicate your code

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java (hunk 1 above)
Comment (Resci) [reply src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: As input i migth get car with drivers and if i ignore them, i will lose data

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java (hunk 2 above)
Comment (Resci) [reply src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: Agree, thanks

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java (hunk 3 above)
Comment (Resci) [reply src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: Видаляє всіх драйверів, яких немає в списку. Якщо видаляти всіх, а потім вставляти погіршується перфоманс

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java (hunk 5 above)
Comment (Resci) [reply src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java:105]: Thanks

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/resources/taxi_schema.sql [hunk 6]
```diff
@@ -1,6 +1,46 @@
...
+-- ----------------------------
+DROP TABLE IF EXISTS `cars`;
+CREATE TABLE `cars`  (
+  `id` bigint(0) UNSIGNED NOT NULL AUTO_INCREMENT,
+  `model` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
+  `manufacturer_id` bigint(0) UNSIGNED NOT NULL,
+  `deleted` bit(1) NOT NULL DEFAULT b'0',
+  PRIMARY KEY (`id`) USING BTREE,
+  INDEX `FK_manufacturer_id`(`manufacturer_id`) USING BTREE,
+  CONSTRAINT `FK_manufacturer_id` FOREIGN KEY (`manufacturer_id`) REFERENCES `manufacturers` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT
```
Comment (KseniiaMakarova) [inline src/main/resources/taxi_schema.sql:39]: let's create the `manufacturers` table first before we run this query

File: src/main/resources/taxi_schema.sql [hunk 7]
```diff
@@ -1,6 +1,46 @@
...
+-- ----------------------------
+DROP TABLE IF EXISTS `cars_drivers`;
+CREATE TABLE `cars_drivers`  (
+  `car_id` bigint(0) UNSIGNED NOT NULL,
+  `driver_id` bigint(0) UNSIGNED NOT NULL,
+  PRIMARY KEY (`car_id`, `driver_id`) USING BTREE,
+  INDEX `driver_id`(`driver_id`) USING BTREE,
+  INDEX `car_id`(`car_id`) USING BTREE,
+  CONSTRAINT `car_id` FOREIGN KEY (`car_id`) REFERENCES `cars` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT,
+  CONSTRAINT `driver_id` FOREIGN KEY (`driver_id`) REFERENCES `drivers` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT
```
Comment (KseniiaMakarova) [inline src/main/resources/taxi_schema.sql:53]: same - create `drivers` first

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setString(1, driver.getName());
+            preparedStatement.setString(2, driver.getLicenceNumber());
+            preparedStatement.setLong(3, driver.getId());
+            Driver old = get(driver.getId()).get();
+            preparedStatement.executeUpdate();
+            return old;
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java]: why do we return the old driver? let's return the updated one

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String selectQuery = "UPDATE drivers SET deleted = true WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(
+                                selectQuery)) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java]:
```suggestion
                        connection.prepareStatement(selectQuery)) {
```

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete driver by id " + id, e);
+        }
+    }
+
+    private Driver parseFromResultSet(ResultSet resultSet) throws SQLException {
+        long driverId = resultSet.getLong("id");
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java]: see checklist - we should use getObject() here

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java (hunk 1 above)
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: here you open one connection inside another - let's move this line below the catch block

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectQuery = "SELECT c.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: why so you use left join here?

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+        long driverId = resultSet.getLong("id");
+        String name = resultSet.getNString("name");
+        String licenseNumber = resultSet.getNString("license_number");
+        Driver driver = new Driver(name, licenseNumber);
+        driver.setId(driverId);
+        return driver;
+    }
+
+    private Car parseCarFromResultSet(ResultSet resultSet) throws SQLException {
+        long manufacturerId = resultSet.getLong("manufacturer_id");
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: use `getObject()` where you need to get Long

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+        String manufacturerName = resultSet.getNString("manufacturer_name");
+        String manufacturerCountry = resultSet.getNString("manufacturer_country");
+        Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+        manufacturer.setId(manufacturerId);
+        long carId = resultSet.getLong("id");
+        String model = resultSet.getNString("model");
+        Car car = new Car(model, manufacturer);
+        car.setId(carId);
+        car.setDrivers(getAllDriversByCarId(carId));
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: here you as well open another connection inside the connection

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            return Optional.of(parseCarFromResultSet(resultSet));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id: " + id, e);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]:
```suggestion
        Car car = null;
        try (Connection connection = ConnectionUtil.getConnection();
                PreparedStatement preparedStatement =
                        connection.prepareStatement(selectQuery)) {
            preparedStatement.setLong(1, id);
            ResultSet resultSet = preparedStatement.executeQuery();
            if (resultSet.next()) {
                car = parseCarFromResultSet(resultSet)); // return only car with set manufacturer here
            }
        } catch (SQLException e) {
            throw new DataProcessingException("Can't get car by id: " + id, e);
        }
        if (car != null) {
             car.setDrivers(getAllDriversByCarId(car.getId());
        }
        return Optional.ofNullable(car);
```

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 15]
```diff
@@ -0,0 +1,257 @@
...
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
+                + " where c.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java:91]: same here - use for each after the catch block

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 16]
```diff
@@ -0,0 +1,257 @@
...
+                             selectQuery)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java:134]: same in this method

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 17]
```diff
@@ -0,0 +1,257 @@
...
+        String selectQuery = "SELECT c.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
+                + " join cars_drivers cd on c.id = cd.car_id"
+                + " join drivers d on cd.driver_id = d.id"
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: do we need different joins here?

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 18]
```diff
@@ -0,0 +1,257 @@
...
+                + " where c.deleted = false and driver_id = ?";
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: we should also check that driver is not deleted

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 19]
```diff
@@ -0,0 +1,257 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String selectQuery = "UPDATE cars SET deleted = true WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(
+                             selectQuery)) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]:
```suggestion
                        connection.prepareStatement(selectQuery)) {
```

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 20]
```diff
@@ -0,0 +1,257 @@
...
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            Car old = get(car.getId()).get();
+            preparedStatement.executeUpdate();
+            updateDriversByCar(car);
+            return old;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]:
```suggestion
            preparedStatement.executeUpdate();
        } catch (SQLException e) {
            throw new DataProcessingException("Can't update car " + car, e);
        }
        deleteAllDrivers(car);
        insertAllDrivers(car);
        return car;
```

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 21]
```diff
@@ -0,0 +1,257 @@
...
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+    }
+
+    private void updateDriversByCar(Car car) {
+        insertAllDrivers(car.getId(), car.getDrivers());
+        deleteAllDriversExceptList(car.getId(), car.getDrivers());
+    }
+
+    private void insertAllDrivers(Long carId, List<Driver> drivers) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: I agree, let's not overcomplicate this at this point please

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 22]
```diff
@@ -0,0 +1,257 @@
...
+                preparedStatement.setLong((i * 2) + 1, carId);
+                preparedStatement.setLong((i * 2) + 2, driver.getId());
+            }
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers " + drivers, e);
+        }
+    }
+
+    private void deleteAllDriversExceptList(Long carId, List<Driver> exceptions) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]:
```suggestion
    private void deleteAllDrivers(Car car) {
```
let's first remove all data that relates to this car, from `cars_drivers` table
and then insert the necessary data to this table - in a different method

File: src/main/java/com/mate/Application.java [hunk 23]
```diff
@@ -41,44 +41,46 @@ public static void main(String[] args) {
...
 
+        tempCar = carService.get(secondCar.getId());
         carService.addDriverToCar(tempDriver, tempCar);
         carService.update(tempCar);
         System.out.println(carService.getAllByDriver(tempDriver.getId()));
+
+        driverService.delete(thirdDriver.getId());
     }
 }
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/Application.java:86]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 24]
```diff
@@ -0,0 +1,248 @@
...
+                        connection.prepareStatement(
+                             insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            insertAllDrivers(car);
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: is another connection opened here inside the current connection?

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 25]
```diff
@@ -0,0 +1,248 @@
...
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectQuery = "SELECT c.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: why left join?

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 26]
```diff
@@ -0,0 +1,248 @@
...
+
+    @Override
+    public List<Car> getAll() {
+        String selectQuery = "SELECT c.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: same

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 27]
```diff
@@ -0,0 +1,248 @@
...
+
+    @Override
+    public boolean delete(Long id) {
+        String selectQuery = "UPDATE cars SET deleted = true WHERE id = ?"
+                + " and deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, id);
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: redundant empty line

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 28]
```diff
@@ -0,0 +1,248 @@
...
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT c.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
+                + " LEFT JOIN cars_drivers cd on c.id = cd.car_id"
+                + " LEFT JOIN drivers d on cd.driver_id = d.id"
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]: let's use inner join here

File: src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java [hunk 29]
```diff
@@ -0,0 +1,248 @@
...
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete drivers " + exceptions, e);
+        }
+    }
+
+    private List<Driver> getAllDriversByCarId(Long carId) {
+        String selectQuery = "SELECT driver_id as id, name, license_number FROM cars_drivers "
+                + "join drivers d on cars_drivers.driver_id = d.id "
+                + "where car_id = ? AND d.deleted = false";
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java]:
```suggestion
        String selectQuery = "SELECT id, name, license_number FROM cars_drivers cd "
                + "JOIN drivers d on cd.driver_id = d.id "
                + "where car_id = ? AND deleted = false";
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
let's use more informative naming for commits

**Pull request merged**